### ISSUE TITLE
Allow specifying index names in POST body

### ISSUE THREAD

**Issue by joschi** [elastic/elasticsearch#26360]
**Describe the feature**:

Most Elasticsearch APIs allow specifying one or more indices in the URI path of an HTTP request, for example the [Search API](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/search-search.html#search-multi-index-type) allows this:

```
GET /kimchy,elasticsearch/tweet/_search?q=tag:wow
```

While this is great, it falls short if the list of indices is very long and exceeds roughly 4 KB, which is the default HTTP initial line length (`http.max_initial_line_length`, [HTTP > Settings](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/modules-http.html#_settings_2)).

Although 4 KB sounds pretty big and "ought to be enough for anybody", there are quite a few real-life scenarios in which that limit can be hit.

**Example scenario:**

Imagine a logging use case with daily rotated indices with the following naming scheme:
```
logstash_YYYY_mm_dd
```

The length of a single index name following this schema is 20 bytes, so it's possible to fit roughly 200 index names in a single query (4096 / 20 = 24.8, not including any other parts of the URI), which isn't much more than half a year worth of logs.

I'm aware that there might be performance problems when including a large number of indices in a query, but given the resources of the Elasticsearch cluster are sufficient, I don't see a reason why not allowing to query a large number of indices.

A workaround for this would be to use wildcards as index name (e. g. `logstash_*`) which might include more indices than necessary and make the performance impact of such a query worse, or use Curator to reindex the documents of multiple indices into a single index (e. g. each week or month) which might cause problems due to the index/shard sizes.

**Comment by joschi** [comment]
#5367 sound remotely related to this issue.

**Comment by javanna** [comment]
Using _msearch is a work-around as it allows to specify indices in the request body for each search request item. Also, index names support wildcard expressions which help trimming the list of indices down. I will mark for discussion though just to make sure.

**Comment by jpountz** [comment]
-1 I don't like having two ways to set index names.

**Comment by colings86** [comment]
I'm also -1 on this as I think there are already various ways to work around this issue including wildcards including wildcard expressions (as mentioned), _msearch (as mentioned), creating aliases for blocks of time (say each week or month) and uses a list of aliases instead of concrete indexes in the search.

Also with the query rewrite and caching functionality added in 5.0 the impact of including more indices than necessary is very much reduced since the query would rewrite to a match no docs query for the extra indices not required in the query range.

**Comment by jasontedor** [comment]
> While this is great, it falls short if the list of indices is very long and exceeds roughly 4 KB, which is the default HTTP initial line length (`http.max_initial_line_length`, [HTTP > Settings](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/modules-http.html#_settings_2)).

For clarity, this can be increased.

**Comment by bleskes** [comment]
@joschi thanks for filing the issue. As you can see there is some resistance to have multiple ways to do the same thing. Can you elaborate some more on how you end up using so many indices? what's your setup?

**Comment by joschi** [comment]
@bleskes The issue with a large list of indices (exceeding the default 4 KB HTTP initial line length) came up in Graylog after switching from a embedded Elasticsearch node to the Elasticsearch HTTP API when users had multiple [index sets](http://docs.graylog.org/en/2.3/pages/configuration/index_model.html) (which you can think of multiple [Logstash Elasticsearch outputs](https://www.elastic.co/guide/en/logstash/5.5/plugins-outputs-elasticsearch.html), each with their own indexing settings) with many indices and they wanted to run search queries across a large number of these indices.

We worked around the issue using index wildcards but using Elasticsearch's [Multi Get API](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/docs-multi-get.html) looks like a better alternative. Thanks for mentioning that!

Refs Graylog2/graylog2-server#4054

**Comment by eskibars** [comment]
Thanks for filing @joschi .  I'm going to close this out since there's a workaround and having multiple ways to do this is likely to lead to confusion if nothing else.  Please do feel free to raise any future issues!

**Comment by nicolem** [comment]
Found this old issue when looking for solutions for a problem in our codebase.  We have so many indexes that we have exceeded the 4K limit on http line length.  Two of our calls were able to be swapped to msearch calls so we can pass the index in there.
We have one search method that then uses scroll for generating sitemaps.  msearch doesn't return a scroll_id, so we can't swap this occurrence for msearch.

Can this issue be revisited, or are there any other workarounds for allowing indexes in the post body and scrolling.
